### ISSUE TITLE
NEAT Algorithms on Ray?

### ISSUE THREAD

**Issue by bionicles** [ray-project/ray#5231]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: 16.04
- **Ray installed from (source or binary)**: binary
- **Ray version**: latest
- **Python version**: 3.6.8 + TF 2.0b1
- **Exact command to reproduce**: n/a

### Describe the problem
We want to play with HyperNEAT algorithms for architecture search of TF2.0 graphs, would it be hard to add this? I thought Tensorflow-NEAT would work but it's built on TF 1.12 ... this seems like a really nice algorithm to have for Ray RLLib

### Source code / logs
https://www.cs.ucf.edu/eplex/ESHyperNEAT/
https://www.youtube.com/watch?v=FUqYNRZTl3U
https://neat-python.readthedocs.io/en/latest/
https://github.com/crisbodnar/TensorFlow-NEAT

**Comment by bionicles** [comment]
good question @richardliaw 

tune is for hyperparameters, but NEAT learns topology parameters in addition to weights, so I think this would count as an RLLib algorithm that just happens to be hierarchical model over model topologies (learn the architecture); so the "policy" for NEAT is just the hypercube network to design the architecture

beyond NEAT, architecture search is a sparse reward situation and RLLib could crush at it

ive had success using NetworkX to make graphs of models and then compiling them recursively pulling outputs. Potentially an architecture search policy could just choose different functions to transform a model graph. If the model crashes, make 'reward' strongly negative

side note: it's easy to make a DataEnv to wrap Tensorflow Datasets, this greatly expands the usefulness of RL

**Comment by arunavo4** [comment]
Any Plans on making this possible ??
